src/Units/KaijuWarsKaiju.java

```diff
@@ -0,0 +1,850 @@
+        }
+      }
+      return events;
+    }
+  }
+  public static class UFOAbducts extends UFO
+  {
+    private static final long serialVersionUID = 1L;
+    public UFOAbducts()
+    {
+      super(null);
+      chargeOnBuildingKill = true;
+    }
+  }
+  public static class UFOMod extends KaijuMoveMod
+  {
+    private static final long serialVersionUID = 1L;
+
+    KaijuStateTracker kaijuTracker;
+    @Override
+    public void registerTrackers(GameInstance gi)
+    {
+      kaijuTracker = StateTracker.instance(gi, KaijuStateTracker.class);
+    }
+
+    @Override
+    public void modifyActionList(UnitContext uc)
+    {
+      if(!kaijuTracker.kaijuAbilityTier.containsKey(uc.unit))
+        return; // Not a registered kaiju
+
+      final KaijuAbilityTier tier = kaijuTracker.kaijuAbilityTier.get(uc.unit);
+      switch (tier)
+      {
+        case ALL:
+          uc.actionTypes.add(new KaijuActions.UFOEMPFactory());
+        case EXTRA:
+          uc.actionTypes.add(new KaijuActions.UFOBeamFactory());
+        case BASIC:
+          break;
+      }
+    }
+
+    @Override
+    public void modifyMovePower(UnitContext uc)
+    {
+      super.modifyMovePower(uc);
+      final KaijuAbilityTier tier = kaijuTracker.kaijuAbilityTier.get(uc.unit);
+      // Nav Computer grants +2 move every turn
+      if( tier != KaijuAbilityTier.BASIC
+          && kaijuTracker.isReady(uc.unit, UFO.class) )
+        uc.movePower += 1;
+    }
+  } //~UFOMod
+
+
+  public static boolean isEnemyRadarScanning(MapMaster map, XYCoord target, Commander affiliation)
+  {
+    boolean isAnyRadar = false;
+    for( XYCoord xyc : Utils.findLocationsInRange(map, target, Radar.PIERCING_VISION) )
+    {
+      Unit resident = map.getResident(xyc);
+      if( null != resident && resident.CO.isEnemy(affiliation)
+          && (resident.model instanceof Radar) )
+      {
+        isAnyRadar = true;
+        break;
+      }
+    }
+    return isAnyRadar;
+  }
+
+
+  /**
+   * Handles cleanup after Kaiju are built:
+   * <p>Gives them the proper HP count, and deletes the port
+   * <p>Stores the creation turn to track abilities
+   * <p>Also handles ability cooldowns
+   */
+  // Should building Kaiju be a different action than building other stuff?
+  public static class KaijuStateTracker extends StateTracker
+  {
+    private static final long serialVersionUID = 1L;
+    public int getTurn() { return game.getCurrentTurn(); }
+
+    // Provide CO energy for studying Kaiju with Radar
+    public GameEventQueue receiveMoveEvent(Unit unit, GamePath unitPath)
+    {
+      if( !(unit.model instanceof Radar) )
+        return null;
+
+      XYCoord end = unitPath.getEndCoord();
+      // Check for any hostile kaiju in range to study
+      for( Unit kaiju : kaijuAbilityTier.keySet() )
+      {
+        if( kaiju.CO.isEnemy(unit.CO) && Radar.PIERCING_VISION >= end.getDistance(unit) )
+        {
+          // TODO: This is definitely wrong
+          unit.CO.modifyAbilityPower(2);
```

Comment:
Hmmm.

Reply:
See my above comment on this.
Though, there probably should be a `ModifyAbilityEnergy` event or such - for animation/audio cues down the line if nothing else.